AdminCore.MailClients.Tests/SMTP/SmtpMailSenderTest.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections.Generic;
+using AdminCore.DTOs.MailMessage;
+using AdminCore.MailClients.Interfaces;
+using AdminCore.MailClients.SMTP;
+using AdminCore.MailClients.SMTP.Interfaces;
+using AutoFixture;
+using AutoFixture.AutoNSubstitute;
+using NSubstitute;
+using Xunit;
+
+namespace AdminCore.MailClients.Tests.SMTP
+{
+    public class SmtpMailSenderTest
+    {
+        private readonly IFixture _fixture = new Fixture().Customize(new AutoNSubstituteCustomization());
+
+        [Fact]
+        public void SendMessages_CallsSendCorrectNoOfTimesGivenListOfMessages()
+        {
+            // Arrange
+            var messageDto = _fixture.Create<MailMessageDto>();
+
+            var smtpClient = Substitute.For<ISmtpClient>();
+            IMailSender smtpSender = GetSmtpMailSender(smtpClient);
+
+            // Act
+            smtpSender.SendMessages(new List<MailMessageDto> {messageDto, messageDto});
+
+            // Assert
+            smtpClient.Received(2).Send(Arg.Any<MailMessageDto>());
+        }
+
+        [Fact]
+        public void SendMessages_CallsSendOncePerEachUniqueMessageGivenListOfMessages()
+        {
+            // Arrange
+            var messageDtoA = _fixture.Create<MailMessageDto>();
```

Comment:
Is it possible to do something like:
```
fixture.CreateMany<MailMessageDto>(3).ToList()
```
Instead of creating them individually?

Reply:
I could, but purpose of the test case is to explicitly test that `Send()` is called *once* on each instance of the DTO object.

Reply:
Cool no worries